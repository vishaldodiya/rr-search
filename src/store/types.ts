import {PinsState} from "./pins/types";
import {DataState} from "./data/types";
import {AnalyticsState} from "./analytics/types";

export interface StoreState {
    pins: PinsState,
    data: DataState,
    analytics: AnalyticsState
};