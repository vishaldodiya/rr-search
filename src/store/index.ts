import {createStore, combineReducers} from "redux";
import * as constants from "../constants";
import DataReducer from "./data/reducer";
import PinsReducer from "./pins/reducer";
import AnalyticsReducer from "./analytics/reducer";
import * as calendar from "../acme-search/calendar.json";
import * as contacts from "../acme-search/contacts.json";
import * as dropbox from "../acme-search/dropbox.json";
import * as slack from "../acme-search/slack.json";
import * as tweet from "../acme-search/tweet.json";

const data = {
    calendar: calendar.calendar,
    contacts: contacts.contacts,
    dropbox: dropbox.dropbox,
    slack: slack.slack,
    tweet: tweet.tweet
};

const reducer = combineReducers({
    pins: PinsReducer,
    data: DataReducer,
    analytics: AnalyticsReducer
});

const store = createStore(reducer);

store.dispatch({
    type: constants.LOAD_DATA,
    list: data
})

store.subscribe(() => console.log(store.getState()));

export default store;