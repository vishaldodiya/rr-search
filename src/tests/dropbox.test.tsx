import * as React from "react";
import Dropbox from "../containers/dropbox";
import {Provider} from "react-redux";
import store from "../store/index";
import * as renderer from 'react-test-renderer';

 
describe('Dropbox', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Dropbox data={
              {
                "id": "12345",
                "path": "/taxes/2016/w2-acme.pdf",
                "title": "\"2016 W2\"",
                "shared_with": [
                  "jane@accountants.com",
                  "spouse@family.org"
                ],
                "created": "2016-02-01",
                "uid": "dropbox-1234"
              }
            }></Dropbox>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});