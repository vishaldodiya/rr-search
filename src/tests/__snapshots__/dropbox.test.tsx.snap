// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropbox snapshot renders 1`] = `
<article
  className="dropbox results__card card"
  id="dropbox-1234"
>
  <div
    className="card__actions"
  >
    <span>
      <i
        className="fab fa-dropbox c--dropbox"
      />
    </span>
    <button
      className="card__pin"
      onClick={[Function]}
    >
      <i
        className="fas fa-thumbtack c--green"
      />
    </button>
  </div>
  <div
    className="dropbox__file"
  >
    <h4>
      "2016 W2"
    </h4>
    <span
      className="dropbox__date"
    >
      January 31, 2016
    </span>
  </div>
  <div
    className="dropbox__shared"
  >
    <i
      className="fas fa-users c--gray"
    />
    <span
      className="card__list"
    >
      <a
        href="mailto:jane@accountants.com"
      >
        jane@accountants.com
      </a>
    </span>
    <span
      className="card__list"
    >
      <a
        href="mailto:spouse@family.org"
      >
        spouse@family.org
      </a>
    </span>
  </div>
  <div
    className="dropbox__share"
  >
    <input
      className="dropbox__share--path c--dark-gray"
      readOnly={true}
      type="text"
      value="https://dropbox.com/taxes/2016/w2-acme.pdf"
    />
    <button
      className="dropbox__share--btn c--dropbox"
      onClick={[Function]}
    >
      Copy link
    </button>
  </div>
</article>
`;
